src/test/java/com/amazon/dlic/auth/ldap/LdapBackendTestNewStyleConfig.java

```diff
@@ -341,14 +341,34 @@ public void testLdapAuthenticationReferral() throws Exception {
 
         final Connection con = LDAPAuthorizationBackend.getConnection(settings, null);
         try {
-            final LdapEntry ref1 = LdapHelper.lookup(con, "cn=Ref1,ou=people,o=TEST", ReturnAttributes.ALL.value());
+            final LdapEntry ref1 = LdapHelper.lookup(con, "cn=Ref1,ou=people,o=TEST", ReturnAttributes.ALL.value(), true);
             Assert.assertEquals("cn=refsolved,ou=people,o=TEST", ref1.getDn());
         } finally {
             con.close();
         }
 
     }
 
+    @Test
+    public void testLdapDontFollowReferrals() throws Exception {
+
+
+        final Settings settings = Settings.builder()
+                .putList(ConfigConstants.LDAP_HOSTS, "localhost:" + ldapPort)
+                .put(ConfigConstants.LDAP_AUTHC_USERSEARCH, "(uid={0})")
+                .put(ConfigConstants.FOLLOW_REFERRALS, false).build();
+
+
+        final Connection con = LDAPAuthorizationBackend.getConnection(settings, null);
+        try {
+            //If following is off then should fail to return the result provided by following
+            final LdapEntry ref1 = LdapHelper.lookup(con, "cn=Ref1,ou=people,o=TEST", ReturnAttributes.ALL.value(), settings.getAsBoolean(ConfigConstants.FOLLOW_REFERRALS, true));
+            Assert.assertEquals("cn=couldNotFollowReferral", ref1.getDn());
```

Comment:
It looks like this test is a mirror of the test above it, lets just check that the expected user isn't found because they are referred.